Sources/Data Model/Experiment.swift

```diff
@@ -63,4 +75,59 @@ extension Experiment {
     var isActivated: Bool {
         return status == .running
     }
+    
+    mutating func serializeAudiences(with audiencesMap: [String: String]) {
+        guard let conditions = audienceConditions else { return }
+        
+        let serialized = conditions.serialized
+        audiences = replaceAudienceIdsWithNames(string: serialized, audiencesMap: audiencesMap)
+    }
+    
+    /// Replace audience ids with audience names
+    ///
+    /// example:
+    /// - string: "(AUDIENCE(1) OR AUDIENCE(2)) AND AUDIENCE(3)"
+    /// - replaced: "(\"us\" OR \"female\") AND \"adult\""
+    ///
+    /// - Parameter string: before replacement
+    /// - Returns: string after replacement
+    func replaceAudienceIdsWithNames(string: String, audiencesMap: [String: String]) -> String {
+        let beginWord = "AUDIENCE("
+        let endWord = ")"
```

Comment:
just curious, is it possible for user to set such characters as values for condition matching?